### PR TITLE
ui: Add eslint presets `plugin:promise/recommended` & `plugin:react-hooks/recommended`

### LINKED ISSUE #1685: ui: Enable recommended eslint presets
- [x] eslint:recommended
- [x] plugin:import/recommended
- [x] plugin:import/typescript
- [x] plugin:react/recommended
- [x] plugin:jest/recommended
- [x] plugin:jest/style
- [x] plugin:jest-dom/recommended
- [x] plugin:@typescript-eslint/recommended
- [ ] plugin:@typescript-eslint/recommended-requiring-type-checking
- [x] plugin:promise/recommended
- [x] plugin:react-hooks/recommended
- [ ] plugin:@next/next/recommended

This can be broken down into multiple PRs

### PR DESCRIPTION
**Pull request by yomete**
Partially resolves #1685

### CODE REVIEW

**Comment by yomete** [comment]
Part two of the eslint recommended presets @manojVivek @monicawoj 

The inclusion of the [allowFinally](https://github.com/eslint-community/eslint-plugin-promise/blob/main/docs/rules/catch-or-return.md#allowfinally) rule `  promise/catch-or-return: [error, {allowFinally: true}]` was needed as we have a couple of places in which we use the ` .finally(fn)` function after the `catch()` function at the end of the promise chain.

The final PR will be about enforcing type checks on the multiple `any`s we have in the code.

File: ui/package.json [hunk 1]
```diff
@@ -78,9 +78,9 @@
     "eslint-plugin-jest-dom": "4.0.3",
     "eslint-plugin-n": "15.6.1",
     "eslint-plugin-prettier": "4.2.1",
-    "eslint-plugin-promise": "6.1.1",
+    "eslint-plugin-promise": "^6.1.1",
```
Comment (maxbrunet) [inline ui/package.json:81]:
```suggestion
    "eslint-plugin-promise": "6.1.1",
```

File: ui/package.json [hunk 2]
```diff
@@ -78,9 +78,9 @@
...
     "eslint-plugin-react": "7.32.1",
-    "eslint-plugin-react-hooks": "4.6.0",
+    "eslint-plugin-react-hooks": "^4.6.0",
```
Comment (maxbrunet) [inline ui/package.json:83]:
```suggestion
    "eslint-plugin-react-hooks": "4.6.0",
```

File: ui/yarn.lock [hunk 3]
```diff
@@ -12984,7 +12984,7 @@ eslint-plugin-prettier@^2.2.0:
     fast-diff "^1.1.1"
     jest-docblock "^21.0.0"
 
-eslint-plugin-promise@6.1.1:
+eslint-plugin-promise@^6.1.1:
```
Comment (maxbrunet) [inline ui/yarn.lock:12987]:
```suggestion
eslint-plugin-promise@6.1.1:
```

File: ui/yarn.lock [hunk 4]
```diff
@@ -12994,7 +12994,7 @@ eslint-plugin-react-hooks@4.3.0:
   resolved "https://registry.yarnpkg.com/eslint-plugin-react-hooks/-/eslint-plugin-react-hooks-4.3.0.tgz#318dbf312e06fab1c835a4abef00121751ac1172"
   integrity sha512-XslZy0LnMn+84NEG9jSGR6eGqaZB3133L8xewQo3fQagbQuGt7a63gf+P1NGKZavEYEC3UXaWEAA/AqDkuN6xA==
 
-eslint-plugin-react-hooks@4.6.0, eslint-plugin-react-hooks@^4.3.0:
+eslint-plugin-react-hooks@4.6.0, eslint-plugin-react-hooks@^4.3.0, eslint-plugin-react-hooks@^4.6.0:
```
Comment (maxbrunet) [inline ui/yarn.lock:12997]:
```suggestion
eslint-plugin-react-hooks@4.6.0, eslint-plugin-react-hooks@^4.3.0:
```

**Review by maxbrunet — APPROVED** [review]
LGTM

Please keep the dev deps pinned. Otherwise Renovate will pin them back

**Pull request merged**